package/service/user_service.go

```diff
@@ -40,6 +49,90 @@ func (us *UserServiceImpl) GetUserByEmail(email string) (*schemas.User, error) {
 	return user, nil
 }
 
+func (us *UserServiceImpl) GetAllUsers(limit, offset int64) ([]schemas.User, error) {
+	tx := us.database.Connect()
+
+	if tx == nil {
+		return nil, ErrDatabaseConnection
+	}
+
+	userModels, err := us.userRepository.GetAllUsers(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	var users []schemas.User
+	for _, userModel := range userModels {
+		users = append(users, *us.modelToSchema(&userModel))
+	}
+
+	// Handle pagination
+	if offset >= int64(len(users)) {
+		return []schemas.User{}, nil
+	}
+
+	end := offset + limit
+	if end > int64(len(users)) {
+		end = int64(len(users))
+	}
+
+	return users[offset:end], nil
+}
+
+func (us *UserServiceImpl) GetUserById(userId int64) (*schemas.User, error){
+	tx := us.database.Connect()
+
+	if tx == nil {
+		return nil, ErrDatabaseConnection
+	}
+
+	userModel, err := us.userRepository.GetUser(tx, userId)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, ErrUserNotFound
+		}
+	}
+
+	user := us.modelToSchema(userModel)
+
+	return user, nil
+}
+
+func (us *UserServiceImpl) EditUser(userId int64, updateInfo *schemas.UserEdit) error {
+	tx := us.database.Connect()
+
+	if tx == nil {
+		return ErrDatabaseConnection
+	}
+
+	tx = tx.Begin()
+    if tx.Error != nil {
+        return tx.Error
+    }
+
+	defer utils.TransactionPanicRecover(tx)
```

Comment:
I dont see why we use this funciton. Could you please explain?

Reply:
In case of a panic this will gracefully rollback the transaction in order to avoid inconsistencies in the database. Another reason is to simply close transaction not to use database resources. Please correct me if Im wrong and suggest different solution

Reply:
Modify according to discussion